### PR TITLE
Convert old-style magics positioners to labels

### PR DESCRIPTION
**Pull request by mrakitin**
@wen-hu will finish these efforts (~40% of devices already converted by @mrakitin).

### CODE REVIEW

**Comment by wen-hu** [comment]
This is the output of old magics:
```
In [67]: %wa                                                                                                                                                                                        
/opt/conda_envs/collection-2019-2.0-csx/lib/python3.6/site-packages/bluesky/magics.py:33: UserWarning: BlueskyMagics.positioners is deprecated. Please use the newer labels feature.
  warnings.warn("BlueskyMagics.positioners is deprecated. "
  Positioner                     Value       Low Limit   High Limit  Offset     
  cryoangle                      24.0        0.0         0.0         AttributeError
  diag2_y                        -0.0001     -100.0      100.0       0.0        
  diag3_y                        -5.3448     -5.3526     -5.3526     -5.3526    
  diag5_y                        -5.0        -4.998      -4.998      -4.998     
  diag6_pid                      0.0         0.0         0.0         AttributeError
  diag6_y                        -11.0       -25.4       -5.4        -5.4       
  epu2_gap                       238.999963  12.0        239.0       AttributeError
  epu2_phase                     3e-05       -24.6       24.6        AttributeError
  es_diag1_y                     -3.8298     -105.0      0.1         0.0        
  eta                            DisconnectedError                                    
  m1a_pit                        6.275034    -20.0       20.0        AttributeError
  m1a_rol                        2.799943    -20.0       20.0        AttributeError
  m1a_x                          -8e-06      -15.0       15.0        AttributeError
  m1a_y                          -2.409544   -10.0       10.0        AttributeError
  m1a_yaw                        5.9e-05     -20.0       20.0        AttributeError
  m1a_z                          -27.619714  -30.0       30.0        AttributeError
  m3a_bdr                        0.25        -10.0       1.0001      0.0        
  m3a_pit                        -0.7198     0.0         0.0         0.0        
  m3a_x                          -0.7499     -10.0       10.0        0.0        
  nanop_bx                       -2.184891   0.0         0.0         0.0        
  nanop_by                       -20.000005  0.0         0.0         -3.26      
  nanop_bz                       -5.739974   0.0         0.0         0.0        
  nanop_tx                       -4.479496   0.0         0.0         0.0        
  nanop_ty                       -20.000001  0.0         0.0         0.0        
  nanop_tz                       4.267627    0.0         0.0         0.0        
  say                            0.0         0.0         0.0         0.0        
  saz                            0.0         0.0         0.0         0.0        
  slt1_xc                        0.0592      -10.0       10.0        -1.0265    
  slt1_xg                        4.0003      -50.0       40.0        22.5996    
  slt1_yc                        0.5717      -100.0      100.0       1.4003     
  slt1_yg                        3.4025      -40.0       60.0        18.0778    
  slt2_xc                        0.0311      -0.069      -0.069      -0.069     
  slt2_xg                        10.1002     14.2938     14.2938     14.2938    
  slt2_yc                        -0.1        0.1515      0.1515      0.1515     
  slt2_yg                        9.0002      14.6062     14.6062     14.6062    
  slt3_x                         -5.6773     -1000.0     1000.0      0.0        
  slt3_y                         -0.2513     0.0         0.0         0.0        
  sx                             3.8         -0.55       14.0        0.0        
  sy                             0.0         0.0         0.0         AttributeError
  sz                             -0.0        0.0         0.0         AttributeError
  tardis_chi                     0.0         0           0           AttributeError
  tardis_delta                   -0.0037     -2.0        178.0       0.0        
  tardis_gamma                   30.0014     -2.9        178.0       0.0        
  tardis_h                       -0.014487   0           0           AttributeError
  tardis_k                       -1.1e-05    0           0           AttributeError
  tardis_l                       0.054064    0           0           AttributeError
  tardis_mu                      0.0         0           0           AttributeError
  tardis_phi                     0.0         0           0           AttributeError
  tardis_theta                   -0.02       -601.0      731.0       0.0        
```

**Comment by wen-hu** [comment]
This is the output of new magics:
'''
In [1]: %wa                                                                                                                                  
tardis
  Positioner                     Value       Low Limit   High Limit  Offset     
  tardis_delta                   -0.0037     -2.0        178.0       0.0        
  tardis_gamma                   30.0013     -2.9        178.0       0.0        
  tardis_h                       -0.133986   0           0           AttributeError
  tardis_k                       -0.000102   0           0           AttributeError
  tardis_l                       0.50002     0           0           AttributeError
  tardis_theta                   -0.02       -601.0      731.0       0.0        

  Local variable name                    Ophyd name (to be recorded as metadata)
  delta                                  tardis_delta                          
  gamma                                  tardis_gamma                          
  tardis.delta                           tardis_delta                          
  tardis.gamma                           tardis_gamma                          
  tardis.h                               tardis_h                              
  tardis.k                               tardis_k                              
  tardis.l                               tardis_l                              
  tardis.theta                           tardis_theta                          
  theta                                  tardis_theta                          

source
  Positioner                     Value       Low Limit   High Limit  Offset     
  epu2_gap                       30.000005   12.0        239.0       AttributeError
  epu2_phase                     5e-06       -24.6       24.6        AttributeError

  Local variable name                    Ophyd name (to be recorded as metadata)
  epu2                                   epu2                                  

optics
  Positioner                     Value       Low Limit   High Limit  Offset     
  diag2_y                        -0.0001     -100.0      100.0       0.0        
  diag3_y                        -5.3448     -5.3526     -5.3526     -5.3526    
  diag5_y                        -5.0        -4.998      -4.998      -4.998     
  diag6_pid                      400.0       0.0         0.0         AttributeError
  diag6_y                        -11.0       -25.4       -5.4        -5.4       
  m1a_pit                        6.275034    -20.0       20.0        AttributeError
  m1a_rol                        2.799943    -20.0       20.0        AttributeError
  m1a_x                          -8e-06      -15.0       15.0        AttributeError
  m1a_y                          -2.409544   -10.0       10.0        AttributeError
  m1a_yaw                        5.9e-05     -20.0       20.0        AttributeError
  m1a_z                          -27.619714  -30.0       30.0        AttributeError
  m3a_bdr                        0.25        -10.0       1.0001      0.0        
  m3a_pit                        -0.72       0.0         0.0         0.0        
  m3a_x                          -0.75       -10.0       10.0        0.0        
  slt1_xc                        0.0593      -10.0       10.0        -1.0265    
  slt1_xg                        3.9996      -50.0       40.0        22.5996    
  slt1_yc                        0.5717      -100.0      100.0       1.4003     
  slt1_yg                        3.4025      -40.0       60.0        18.0778    
  slt2_xc                        0.0311      -0.069      -0.069      -0.069     
  slt2_xg                        10.1002     14.2938     14.2938     14.2938    
  slt2_yc                        -0.1        0.1515      0.1515      0.1515     
  slt2_yg                        9.0002      14.6062     14.6062     14.6062    
  slt3_x                         -5.6772     -1000.0     1000.0      0.0        
  slt3_y                         -0.2512     0.0         0.0         0.0        

  Local variable name                    Ophyd name (to be recorded as metadata)
  diag2_y                                diag2_y                               
  diag3_y                                diag3_y                               
  diag5_y                                diag5_y                               
  diag6_pid                              diag6_pid                             
  diag6_y                                diag6_y                               
  m1a                                    m1a                                   
  m3a                                    m3a                                   
  slt1                                   slt1                                  
  slt2                                   slt2                                  
  slt3                                   slt3                                  

motors
  Positioner                     Value       Low Limit   High Limit  Offset     
  cryoangle                      24.0        0.0         0.0         AttributeError
  es_diag1_y                     -3.8298     -105.0      0.1         0.0        
  nanop_bx                       -2.193926   0.0         0.0         0.0        
  nanop_by                       -20.000001  0.0         0.0         -3.26      
  nanop_bz                       -5.739798   0.0         0.0         0.0        
  nanop_tx                       -4.479498   0.0         0.0         0.0        
  nanop_ty                       -20.000008  0.0         0.0         0.0        
  nanop_tz                       4.267594    0.0         0.0         0.0        
  say                            0.0         0.0         0.0         0.0        
  saz                            0.0         0.0         0.0         0.0        
  sx                             3.8         -0.55       14.0        0.0        
  sy                             0.0         0.0         0.0         AttributeError
  sz                             -0.0        0.0         0.0         AttributeError

  Local variable name                    Ophyd name (to be recorded as metadata)
  cryoangle                              cryoangle                             
  es_diag1_y                             es_diag1_y                            
  nanop                                  nanop                                 
  say                                    say                                   
  saz                                    saz                                   
  sx                                     sx                                    
  sy                                     sy                                    
  sz                                     sz  
'''

File: profile_collection/startup/csx1/startup/settings.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
 sd.baseline = [theta, delta, gamma,
                sx, say, saz,
                cryoangle, sy, sz,
-               epu1, epu2,
+               #epu1, epu2,
+               epu2,
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/settings.py]: @wen-hu, do you want both epu's here, or just `epu2`? I think we commented the two of them for tests...

File: profile_collection/startup/csx1/startup/endstation.py [hunk 2]
```diff
@@ -17,12 +17,12 @@
 
 # Sample positions
 
-sx = EpicsMotor('XF:23ID1-ES{Dif-Ax:X}Mtr', name='sx')
-sy = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SY}', name='sy')
-sz = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SZ}', name='sz')
-say = EpicsMotor('XF:23ID1-ES{Dif-Ax:Y}Mtr', name='say')
-saz = EpicsMotor('XF:23ID1-ES{Dif-Ax:Z}Mtr', name='saz')
-cryoangle = Cryoangle('', name='cryoangle')
+sx = EpicsMotor('XF:23ID1-ES{Dif-Ax:X}Mtr', name='sx',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
sx = EpicsMotor('XF:23ID1-ES{Dif-Ax:X}Mtr', name='sx', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/endstation.py [hunk 3]
```diff
@@ -17,12 +17,12 @@
...
+sy = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SY}', name='sy',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
sy = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SY}', name='sy', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/endstation.py [hunk 4]
```diff
@@ -17,12 +17,12 @@
...
+sz = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SZ}', name='sz',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
sz = SamplePosVirtualMotor('XF:23ID1-ES{Dif-Ax:SZ}', name='sz', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/endstation.py [hunk 5]
```diff
@@ -17,12 +17,12 @@
...
+say = EpicsMotor('XF:23ID1-ES{Dif-Ax:Y}Mtr', name='say',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
say = EpicsMotor('XF:23ID1-ES{Dif-Ax:Y}Mtr', name='say', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/endstation.py [hunk 6]
```diff
@@ -17,12 +17,12 @@
...
+saz = EpicsMotor('XF:23ID1-ES{Dif-Ax:Z}Mtr', name='saz',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
saz = EpicsMotor('XF:23ID1-ES{Dif-Ax:Z}Mtr', name='saz', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/endstation.py [hunk 7]
```diff
@@ -31,7 +31,7 @@
 
 # Diagnostic Axis
 
-es_diag1_y = EpicsMotor('XF:23ID1-ES{Diag:1-Ax:Y}Mtr', name='es_diag1_y')
+es_diag1_y = EpicsMotor('XF:23ID1-ES{Diag:1-Ax:Y}Mtr', name='es_diag1_y',labels=['motors'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/endstation.py]:
```suggestion
es_diag1_y = EpicsMotor('XF:23ID1-ES{Diag:1-Ax:Y}Mtr', name='es_diag1_y', labels=['motors'])
```

File: profile_collection/startup/csx1/startup/optics.py [hunk 8]
```diff
@@ -11,7 +11,7 @@
 
 # M1A, M1B1, M1B2
 
-m1a = FMBHexapodMirror('XF:23IDA-OP:1{Mir:1', name='m1a')
+m1a = FMBHexapodMirror('XF:23IDA-OP:1{Mir:1', name='m1a',labels=['optics'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/optics.py]:
```suggestion
m1a = FMBHexapodMirror('XF:23IDA-OP:1{Mir:1', name='m1a', labels=['optics'])
```

File: profile_collection/startup/csx1/startup/optics.py [hunk 9]
```diff
@@ -20,24 +20,24 @@
 
 # M3A Mirror
 
-m3a = M3AMirror('XF:23ID1-OP{Mir:3',  name='m3a')
+m3a = M3AMirror('XF:23ID1-OP{Mir:3',  name='m3a',labels=['optics'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/optics.py]:
```suggestion
m3a = M3AMirror('XF:23ID1-OP{Mir:3',  name='m3a', labels=['optics'])
```

File: profile_collection/startup/csx1/startup/optics.py [hunk 10]
```diff
@@ -20,24 +20,24 @@
...
 
 # Slits
 
-slt1 = SlitsGapCenter('XF:23ID1-OP{Slt:1', name='slt1')
-slt2 = SlitsGapCenter('XF:23ID1-OP{Slt:2', name='slt2')
-slt3 = SlitsXY('XF:23ID1-OP{Slt:3', name='slt3')
+slt1 = SlitsGapCenter('XF:23ID1-OP{Slt:1', name='slt1',labels=['optics'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/optics.py]:
```suggestion
slt1 = SlitsGapCenter('XF:23ID1-OP{Slt:1', name='slt1', labels=['optics'])
```

File: profile_collection/startup/csx1/startup/optics.py [hunk 11]
```diff
@@ -20,24 +20,24 @@
...
+slt2 = SlitsGapCenter('XF:23ID1-OP{Slt:2', name='slt2',labels=['optics'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/optics.py]:
```suggestion
slt2 = SlitsGapCenter('XF:23ID1-OP{Slt:2', name='slt2', labels=['optics'])
```

File: profile_collection/startup/csx1/startup/optics.py [hunk 12]
```diff
@@ -20,24 +20,24 @@
...
+slt3 = SlitsXY('XF:23ID1-OP{Slt:3', name='slt3',labels=['optics'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/optics.py]:
```suggestion
slt3 = SlitsXY('XF:23ID1-OP{Slt:3', name='slt3', labels=['optics'])
```

File: profile_collection/startup/csx1/startup/tardis.py [hunk 13]
```diff
@@ -20,17 +20,17 @@ def connected(self):
...
-    h = Cpt(PseudoSingle, '')
-    k = Cpt(PseudoSingle, '')
-    l = Cpt(PseudoSingle, '')
+    h = Cpt(PseudoSingle, '', labels=['tardis'])
+    k = Cpt(PseudoSingle, '', labels=['tardis'])
+    l = Cpt(PseudoSingle, '', labels=['tardis'])
 
-    theta = Cpt(EpicsMotor, 'XF:23ID1-ES{Dif-Ax:Th}Mtr')
-    mu = Cpt(NullMotor)
+    theta = Cpt(EpicsMotor, 'XF:23ID1-ES{Dif-Ax:Th}Mtr',labels=['tardis'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/tardis.py]:
```suggestion
    theta = Cpt(EpicsMotor, 'XF:23ID1-ES{Dif-Ax:Th}Mtr', labels=['tardis'])
```

File: profile_collection/startup/csx1/startup/tardis.py [hunk 14]
```diff
@@ -20,17 +20,17 @@ def connected(self):
...
+    mu = Cpt(NullMotor,labels=['tardis'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/tardis.py]:
```suggestion
    mu = Cpt(NullMotor, labels=['tardis'])
```

File: profile_collection/startup/csx1/startup/tardis.py [hunk 15]
```diff
@@ -20,17 +20,17 @@ def connected(self):
...
 
-    chi =   Cpt(NullMotor)
-    phi =   Cpt(NullMotor)
-    delta = Cpt(EpicsMotor, 'XF:23ID1-ES{Dif-Ax:Del}Mtr')
-    gamma = Cpt(EpicsMotor, 'XF:23ID1-ES{Dif-Ax:Gam}Mtr')
+    chi =   Cpt(NullMotor,labels=['tardis'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/tardis.py]:
```suggestion
    chi =   Cpt(NullMotor, labels=['tardis'])
```

File: profile_collection/startup/csx1/startup/tardis.py [hunk 16]
```diff
@@ -20,17 +20,17 @@ def connected(self):
...
+    phi =   Cpt(NullMotor,labels=['tardis'])
```
Comment (mrakitin) [inline profile_collection/startup/csx1/startup/tardis.py]:
```suggestion
    phi =   Cpt(NullMotor, labels=['tardis'])
```

**Comment by mrakitin** [comment]
Checked in person with @wen-hu.

**Comment by mrakitin** [comment]
The latest version is:
```py
In [1]: %wa
tardis
  Positioner                     Value       Low Limit   High Limit  Offset
  tardis_chi                     0.0         0           0           AttributeError
  tardis_delta                   -0.0037     -2.0        178.0       0.0
  tardis_gamma                   30.0013     -2.9        178.0       0.0
  tardis_h                       -0.133986   0           0           AttributeError
  tardis_k                       -0.000102   0           0           AttributeError
  tardis_l                       0.500019    0           0           AttributeError
  tardis_mu                      0.0         0           0           AttributeError
  tardis_phi                     0.0         0           0           AttributeError
  tardis_theta                   -0.02       -601.0      731.0       0.0

  Local variable name                    Ophyd name (to be recorded as metadata)
  delta                                  tardis_delta
  gamma                                  tardis_gamma
  tardis.chi                             tardis_chi
  tardis.delta                           tardis_delta
  tardis.gamma                           tardis_gamma
  tardis.h                               tardis_h
  tardis.k                               tardis_k
  tardis.l                               tardis_l
  tardis.mu                              tardis_mu
  tardis.phi                             tardis_phi
  tardis.theta                           tardis_theta
  theta                                  tardis_theta

source
  Positioner                     Value       Low Limit   High Limit  Offset
  epu2_gap                       29.999982   12.0        239.0       AttributeError
  epu2_phase                     -2.7e-05    -24.6       24.6        AttributeError

  Local variable name                    Ophyd name (to be recorded as metadata)
  epu2                                   epu2

optics
  Positioner                     Value       Low Limit   High Limit  Offset
  diag2_y                        -0.0001     -100.0      100.0       0.0
  diag3_y                        -5.3448     -5.3526     -5.3526     -5.3526
  diag5_y                        -5.0        -4.998      -4.998      -4.998
  diag6_pid                      400.0       0.0         0.0         AttributeError
  diag6_y                        -11.0       -25.4       -5.4        -5.4
  m1a_pit                        6.275034    -20.0       20.0        AttributeError
  m1a_rol                        2.799943    -20.0       20.0        AttributeError
  m1a_x                          -8e-06      -15.0       15.0        AttributeError
  m1a_y                          -2.409544   -10.0       10.0        AttributeError
  m1a_yaw                        5.9e-05     -20.0       20.0        AttributeError
  m1a_z                          -27.619714  -30.0       30.0        AttributeError
  m3a_bdr                        0.25        -10.0       1.0001      0.0
  m3a_pit                        -0.72       0.0         0.0         0.0
  m3a_x                          -0.75       -10.0       10.0        0.0
  slt1_xc                        0.0593      -10.0       10.0        -1.0265
  slt1_xg                        3.9996      -50.0       40.0        22.5996
  slt1_yc                        0.5717      -100.0      100.0       1.4003
  slt1_yg                        3.4025      -40.0       60.0        18.0778
  slt2_xc                        0.0311      -0.069      -0.069      -0.069
  slt2_xg                        10.1002     14.2938     14.2938     14.2938
  slt2_yc                        -0.1        0.1515      0.1515      0.1515
  slt2_yg                        9.0002      14.6062     14.6062     14.6062
  slt3_x                         -5.6772     -1000.0     1000.0      0.0
  slt3_y                         -0.2512     0.0         0.0         0.0

  Local variable name                    Ophyd name (to be recorded as metadata)
  diag2_y                                diag2_y
  diag3_y                                diag3_y
  diag5_y                                diag5_y
  diag6_pid                              diag6_pid
  diag6_y                                diag6_y
  m1a                                    m1a
  m3a                                    m3a
  slt1                                   slt1
  slt2                                   slt2
  slt3                                   slt3

motors
  Positioner                     Value       Low Limit   High Limit  Offset
  cryoangle                      24.0        0.0         0.0         AttributeError
  es_diag1_y                     -3.8298     -105.0      0.1         0.0
  nanop_bx                       -2.193953   0.0         0.0         0.0
  nanop_by                       -20.000001  0.0         0.0         -3.26
  nanop_bz                       -5.739828   0.0         0.0         0.0
  nanop_tx                       -4.479523   0.0         0.0         0.0
  nanop_ty                       -20.000005  0.0         0.0         0.0
  nanop_tz                       4.267621    0.0         0.0         0.0
  say                            0.0         0.0         0.0         0.0
  saz                            0.0         0.0         0.0         0.0
  sx                             3.8         -0.55       14.0        0.0
  sy                             0.0         0.0         0.0         AttributeError
  sz                             -0.0        0.0         0.0         AttributeError

  Local variable name                    Ophyd name (to be recorded as metadata)
  cryoangle                              cryoangle
  es_diag1_y                             es_diag1_y
  nanop                                  nanop
  say                                    say
  saz                                    saz
  sx                                     sx
  sy                                     sy
  sz                                     sz
```

**Pull request merged**